src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexQueryPlan.java

```diff
@@ -74,36 +71,108 @@ private StorageAttachedIndexQueryPlan(ColumnFamilyStore cfs,
     @Nullable
     public static StorageAttachedIndexQueryPlan create(ColumnFamilyStore cfs,
                                                        TableQueryMetrics queryMetrics,
-                                                       Set<StorageAttachedIndex> indexes,
+                                                       Set<StorageAttachedIndex> allIndexes,
                                                        RowFilter rowFilter)
     {
-        ImmutableSet.Builder<Index> selectedIndexesBuilder = ImmutableSet.builder();
-        List<RowFilter.Expression> acceptedExpressions = new ArrayList<>();
+        // collect the indexes that can be used with the provided row filter
```

Comment:
At some point we could pass this list to the `Index.Searcher` so it can select between the indexes identified here. But I'd prefer to do that in a separate PR and focus this one on solving the reported bugs.